### PR TITLE
rsyslog - support action retry, max wait

### PR DESCRIPTION
**Pull request by richm**
This makes rsyslog have the same behavior as fluentd
```
        retry_max_interval "#{ENV['ES_RETRY_WAIT'] || '300'}"
        retry_forever true
```
Ref https://jira.coreos.com/browse/LOG-399

### CODE REVIEW

File: files/rsyslog/55-elasticsearch.conf [hunk 1]
```diff
@@ -61,6 +61,8 @@ ruleset(name="try_es") {
             queue.maxFileSize=`echo $ES_QUEUE_MAXFILESIZE`
             queue.checkpointInterval=`echo $ES_QUEUE_CHECKPOINTINTERVAL`
             queue.syncqueuefiles="on"
+            action.resumeRetryCount="-1"
+            action.resumeIntervalMax=`echo $ES_RETRY_WAIT`
```
Comment (nhosoi) [inline files/rsyslog/55-elasticsearch.conf]: I cannot find resumeIntervalMax but resumeInterval in the doc and the sources...  Could you point me to the reference?

Comment (richm) [reply files/rsyslog/55-elasticsearch.conf]: good catch - `action.resumeIntervalMax` is not supported in our version of rsyslog, even though it is listed on the page, and doesn't say `new in 8.xx version` - https://www.rsyslog.com/doc/v8-stable/configuration/actions.html#general-action-parameters

**Comment by richm** [comment]
please review and lgtm

**Comment by ewolinetz** [comment]
not sure why you put hold, but lgtm

**Comment by richm** [comment]
the hold is because I thought we were not supposed to merge this for 4.1 - @bparees @jcantrill please confirm

**Comment by bparees** [comment]
confirmed.  unless not merging this leaves us exposed to some GA functionality breakage due to the pieces of this that already landed.

but if it only relates to enabling what was going to be tech preview in 4.1, then it should be held for 4.2 instead.

**Comment by jcantrill** [comment]
@richm i changed the base.  is there a jira card we can ref here

**Comment by richm** [comment]
> @richm i changed the base. is there a jira card we can ref here

there is now - https://jira.coreos.com/browse/LOG-399

**Comment by bparees** [comment]
why is this merging when the exception was not approved?

**Comment by bparees** [comment]
oh sorry, i see it's a 4.2 branch
/hold cancel

**Pull request merged**